WBooks/Book Details/BookDetailsViewController.swift

```diff
@@ -36,47 +36,56 @@ class BookDetailsViewController: UIViewController {
     
     private func setBookDetails() {
         _view.setBook(bookViewModel: _viewModel.bookViewModel)
-        _viewModel.loadRents(onSuccess: onLoadRentsSuccess)
-    }
-    
-    func onLoadRentsSuccess(rents: [Rent]) {
-        _viewModel.onLoadRentsSuccess(rents: rents)
-        _view.setAvailability(status: _viewModel.status)
+        _viewModel.status.producer.startWithValues { [weak self] status in
+            guard let this = self else {
+                return
+            }
+            this._view.setAvailability(status: status)
+        }
+        _viewModel.loadRents()
     }
     
     func setRentButton() {
+        //Rent button action
         _view.rent.reactive.controlEvents(.touchUpInside).observeValues { [weak self] _ in
             self?.rent()
         }
+        
+        //Actions when rent request is completed
+        _viewModel.rentState.producer.startWithValues { [weak self] state in
+            guard let this = self else {
+                return
+            }
+            switch state {
+            case .rentError:
+                this._view.processingRentAlert.dismiss(animated: true, completion: nil)
+                let alert = UIAlertController(alertViewModel: ErrorAlertViewModel(title: "ERROR".localized(), message: "RENT_ERROR".localized(), dismissButtonTitle: "ACCEPT".localized()))
+                this.present(alert, animated: true, completion: nil)
+            case .rentSuccess:
+                this._view.processingRentAlert.dismiss(animated: true, completion: nil)
+                let alert = UIAlertController(alertViewModel: ErrorAlertViewModel(title: "THANKS".localized(), message: "RENT_COMPLETED".localized(), dismissButtonTitle: "ACCEPT".localized()))
+                this.present(alert, animated: true, completion: nil)
+                this._view.setAvailability(status: this._viewModel.status.value)
+            case .rentSleep:
+                break
+            }
+        }
     }
-    
+
     func rent() {
-        switch _viewModel.status {
+        switch _viewModel.status.value {
         case .available:
-            let alert = UIAlertController(alertViewModel: ErrorAlertViewModel(title: "THANKS".localized(), message: "PROCESSING_RENT".localized(), dismissButtonTitle: "ACCEPT".localized()))
-            self.present(alert, animated: true, completion: nil)
-            _viewModel.rentBook(onSuccess: onBookRentSuccess, onError: onBookRentError)
+            present(_view.processingRentAlert, animated: true, completion: nil)
```

Comment:
Why did you set this alert in the view?

Reply:
It better belongs to the controller now that you point this out. I'll change it.